src/GafferUIModule/PathListingWidgetBinding.cpp

```diff
@@ -1369,7 +1369,12 @@ class PathModel : public QAbstractItemModel
 
 					for( auto it = children.begin(), eIt = children.end(); it != eIt; ++it )
 					{
-						auto oldIt = oldChildMap.find( (*it)->names().back() );
+						const Path::Names &names = (*it)->names();
+						if( !names.size() )
+						{
+							continue;
+						}
+						auto oldIt = oldChildMap.find( names.back() );
```

Comment:
> is there some way we could include a unit test that exercises it?

I'll look into it - I expect there is a way.

> If so, should we be endeavouring to make PathListingWidget more savvy to the multiple-roots situation so that it can show them all?

I'll double check this evening and try some non-standard scenarios, but I think not showing the roots in this case is desirable and probably not practically possible anyways. The comparison to Linux roots is maybe not the best for UNC paths. They operate more like network URLs like `//server.name/shareName/directoryName/`, or `//1.2.3.4/shareName/directoryName` and as far as I know there's not "registration" system in Windows to store available servers (roots in our case). So in our nomenclature, a root could become available or drop out with no notification to the Windows OS or us, until we try and query it again when the query would either succeed or fail.

I suppose we could cache UNC path roots after they've been visited once, and then we could present those as roots when the user has typed in `//`.